pyccel/ast/numpyext.py

```diff
@@ -89,6 +89,75 @@
     'NumpyArange'
 )
 
+#=======================================================================================
+class NumpyComplex(PythonComplex):
+    """ Represents a call to numpy.complex() function.
+    """
+
+class NumpyComplex64(NumpyComplex):
+    _precision = dtype_registry['complex64'][1]
+
+class NumpyComplex128(NumpyComplex):
+    _precision = dtype_registry['complex128'][1]
+
+#=======================================================================================
+class NumpyFloat(PythonFloat):
+    """ Represents a call to numpy.float() function.
+    """
+
+class NumpyFloat32(NumpyFloat):
+    """ Represents a call to numpy.float32() function.
+    """
+    _precision = dtype_registry['float32'][1]
+
+class NumpyFloat64(NumpyFloat):
+    """ Represents a call to numpy.float64() function.
+    """
+    _precision = dtype_registry['float64'][1]
+
+#=======================================================================================
+# TODO [YG, 13.03.2020]: handle case where base != 10
+class NumpyInt(PythonInt):
+    """ Represents a call to numpy.int() function.
+    """
+    def __new__(cls, arg=None, base=10):
+        return PythonInt.__new__(cls, arg)
+
+class NumpyInt32(NumpyInt):
+    """ Represents a call to numpy.int32() function.
+    """
+    _precision = dtype_registry['int32'][1]
+
+class NumpyInt64(NumpyInt):
+    """ Represents a call to numpy.int64() function.
+    """
+    _precision = dtype_registry['int64'][1]
+
+#==============================================================================
+class NumpyReal(PythonReal):
+    """Represents a call to  numpy.real for code generation.
+
+    > a = 1+2j
+    > np.real(a)
+    1.0
+    """
+
+#==============================================================================
+PrecisionToDtype = {
```

Comment:
I don't think this name describes what this dictionary does. `PrecisionToDtype` implies that the key is a `precision` and the result is the `dtype`, but here the result is a casting function

Reply:
i agree.